### PR TITLE
Added C++ Binary Search with fewer comparisons

### PR DESCRIPTION
**Pull request by Pl4gue**
**Changes:**
Added C++ Binary Search with fewer comparisons ( only 1 comparison in while loop instead of 2)

### CODE REVIEW

File: code/search/binary_search/binary_search_2.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
+/* Part of Cosmos by OpenGenus Foundation */
+
+#include <vector>
+#include <stdlib.h>
```
Comment (arnavb) [inline code/search/binary_search/binary_search_2.cpp]: Use `cstdlib`

**Comment by arnavb** [comment]
Don't forget to read the C++ Style Guide!

File: code/search/binary_search/binary_search_2.cpp (hunk 1 above)
Comment (Pl4gue) [reply code/search/binary_search/binary_search_2.cpp]: Done

File: code/search/binary_search/binary_search_2.cpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    cout << "Array (sorted) : ";
+    sort(v.begin(),v.end());    
+    printarr(v);
+    
+    cout << "Search for (input search key) : ";
+    int key;
+    cin >> key;   
+    
+    cout << "Found " << key << " at index "<< binary_search(v,key) << endl;
+}
```
Comment (yatharthshahjpr) [inline code/search/binary_search/binary_search_2.cpp]: Please add a new line at the last ! 👍

Comment (Pl4gue) [reply code/search/binary_search/binary_search_2.cpp]: What exactly do you mean by that? Just adding an empty line at the end?

Comment (arnavb) [reply code/search/binary_search/binary_search_2.cpp]: @yatharthshahjpr There already is a newline, otherwise Github would say so.

**Comment by Pl4gue** [comment]
When are you going to merge this? 😄

**Comment by AdiChat** [comment]
Looks great 😍 

There has been some delay. 😅 It's never late to merge a great contribution. 👍 
Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**